src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -147,50 +164,129 @@ public ExtractInterfaceResult ExtractInterfaceFromAnalyzedType(ExtractInterfaceT
                 typeParameters: GetTypeParameters(refactoringResult.TypeToExtractFrom, extractInterfaceOptions.IncludedMembers),
                 members: CreateInterfaceMembers(extractInterfaceOptions.IncludedMembers));
 
-            var interfaceDocumentId = DocumentId.CreateNewId(refactoringResult.DocumentToExtractFrom.Project.Id, debugName: extractInterfaceOptions.FileName);
+            switch (extractInterfaceOptions.Location)
+            {
+                case ExtractInterfaceOptionsResult.ExtractLocation.NewFile:
+                    var containingNamespaceDisplay = GetContainingNamespaceDisplay(refactoringResult.TypeToExtractFrom, refactoringResult.DocumentToExtractFrom.Project.CompilationOptions);
+                    return await ExtractInterfaceToNewFileAsync(
+                        solution: solution,
+                        containingNamespaceDisplay: containingNamespaceDisplay,
+                        extractedInterfaceSymbol: extractedInterfaceSymbol,
+                        refactoringResult: refactoringResult,
+                        extractInterfaceOptions: extractInterfaceOptions,
+                        cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                case ExtractInterfaceOptionsResult.ExtractLocation.SameFile:
+                    return await ExtractInterfaceToSameFileAsync(
+                        solution: solution,
+                        extractedInterfaceSymbol: extractedInterfaceSymbol,
+                        refactoringResult: refactoringResult,
+                        extractInterfaceOptions: extractInterfaceOptions,
+                        cancellationToken: cancellationToken).ConfigureAwait(false);
+
+                default: throw new InvalidOperationException($"Unable to extract interface for operation of type {extractInterfaceOptions.GetType()}");
+            }
+        }
 
-            var unformattedInterfaceDocument = GetUnformattedInterfaceDocument(
+        private async Task<ExtractInterfaceResult> ExtractInterfaceToNewFileAsync(
+            Solution solution, string containingNamespaceDisplay, INamedTypeSymbol extractedInterfaceSymbol,
+            ExtractInterfaceTypeAnalysisResult refactoringResult, ExtractInterfaceOptionsResult extractInterfaceOptions, CancellationToken cancellationToken)
+        {
+            var symbolMapping = await CreateSymbolToDeclarationAnnotationMapAsync(
+                extractInterfaceOptions.IncludedMembers,
                 solution,
+                refactoringResult.TypeNode,
+                cancellationToken).ConfigureAwait(false);
```

Comment:
nit: returns a SymbolMapping right?  so CreateSymbolMapping would be a good name.